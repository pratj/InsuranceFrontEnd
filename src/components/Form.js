import React, {useEffect, useState} from 'react'
import RenderForm from './RenderForm';
import axios from 'axios'
import { useHistory } from "react-router";
import useGeoLocation from '../geolocation/useGeoLocation'

function Form({cardInfo, setOpenPopup}) {
        
    const [formFields, setFormFields] = useState()
    const history = useHistory();
    const geoLocation= useGeoLocation();

    const onSubmit = (data) => {

      const userLocation= geoLocation.loaded ? (geoLocation): "Location data not available yet."
      const finalData = {category: cardInfo.category, product: cardInfo.product, formData: data, userLocation:userLocation}
      axios.post("http://localhost:9090/api/response", finalData).then((response) => {
        const quoteData={category: cardInfo.category,product: cardInfo.product, quoteData: response.data, userLocation:userLocation} 
        redirectToPath(quoteData)
      })
      setOpenPopup(false)
    }

    function redirectToPath(quoteData) {
      history.push("/quote", {quoteData: JSON.stringify(quoteData)})
    }
  
    useEffect(() => {
        axios.get(`http://localhost:9090/api/config/category/${cardInfo.category}/product/${cardInfo.product}`).then((response) => {
            setFormFields(response.data[0].fields)
        })    
    }, [cardInfo.category, cardInfo.product])


    return (

      <div>
        {typeof formFields !== 'undefined' && 
          <RenderForm formFields={formFields} onSubmit={onSubmit}/>
        }    
      </div>
    )     
}

export default Form
