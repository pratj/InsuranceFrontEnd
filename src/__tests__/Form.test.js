import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Form from "../components/Form";

const cardInfo = {
  category: "Motor Insurance",
  product: "Car Insurance",
};

const setOpenPopup = true;

test("renders form component", () => {
  shallow(<Form cardInfo={cardInfo} setOpenPopup={setOpenPopup} />);
});

test("renders form component with parent div", () => {
  const wrapper = shallow(
    <Form cardInfo={cardInfo} setOpenPopup={setOpenPopup} />
  );
  expect(wrapper.find('[data-test="form"]').length).toBe(1);
});

// test("accepts form props", () => {
//   const wrapper = mount(
//     <Form cardInfo={cardInfo} setOpenPopup={setOpenPopup} />
//   );
//   expect(wrapper.props().cardInfo).toEqual(cardInfo);
// });

test("Form snapshots", () => {
  const formTree = shallow(
    <Form cardInfo={cardInfo} setOpenPopup={setOpenPopup} />
  );
  expect(toJson(formTree)).toMatchSnapshot();
});