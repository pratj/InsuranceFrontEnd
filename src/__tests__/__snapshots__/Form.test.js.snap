// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshots 1`] = `
<div
  className="form"
  data-test="form"
/>
`;
